### PR TITLE
TASK-58038: Add based swagger rest documentation for social and gatein-portal

### PR DESCRIPTION
**Pull request by hakermi**
Add based swagger rest documentation for social and gatein-portal

### CODE REVIEW

File: docs/API.rst [hunk 1]
```diff
@@ -119,6 +119,8 @@ integrate eXo services into their applications.
 See `REST API Documentation 5.0 <../../../PLF50/rest-api>`__ for more
 details.
 
+See :ref:`eXo Rest APIs Swagger Documented <SWAGGER.REST.APIS>` for more details.
```
Comment (ahamdi) [inline docs/API.rst]:
```suggestion
See :ref:`eXo Rest APIs Documentation <SWAGGER.REST.APIS>` for more details.
```

File: docs/_static/js/DocsUtils.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            error: function (error) {
+                reject(error);
+            }
+        })
+    })
+}
+
+function swaggerUi(options) {
+    const url = options.url + options.restModule + '/' + options.version;
+    getBuildVersion(url).then(buildVersion => {
```
Comment (ahamdi) [inline docs/_static/js/DocsUtils.js]: this call is needed when it contains '-SNAPSHOT' String, could we execute it just then

File: docs/Swagger_Rest_Apis.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ :ref:`Social Rest API <SOCIAL_REST_API>`
+
+ :ref:`Gatein-Portal Rest API <GATEIN_PORTAL_REST_API>`
+
+
+.. _SOCIAL_REST_API:
+
+  **Social Rest Api**
+
+  Swagger ui based rest documentation for social.
```
Comment (ahamdi) [inline docs/Swagger_Rest_Apis.rst]:
```suggestion
  The eXo Social REST APIs provide service endpoints for listing, creating, updating and deleting entities related to Social functionality.
  Following the list of eXo Social API operations : 
```

File: docs/Swagger_Rest_Apis.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+ .. raw:: html
+   :file: embed/social-docs-swagger.html
+
+.. _GATEIN_PORTAL_REST_API:
+
+  **Gatein-portal Rest Api**
+
+  Swagger ui based rest documentation for Gatein-Portal.
```
Comment (ahamdi) [inline docs/Swagger_Rest_Apis.rst]:
```suggestion
  The eXo Portal REST APIs provide service endpoints for listing, creating, updating and deleting entities related to Portal layer functionality.
  Following the list of eXo Portal API operations : 
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: docs/_static/js/DocsUtils.js (hunk 2 above)
Comment (hakermi) [reply docs/_static/js/DocsUtils.js]: done

**Review by ahamdi — APPROVED** [review]

**Pull request merged**